### DISCUSSION TITLE
An actual array instead of an array datatype in OCAFile produced schema bundle

### DISCUSSION THREAD

**Discussion by carlyh-micb** [the-human-colossus-foundation/oca-spec#66] [General]
Looking at an OCA Schema bundle that I created using Swagger API (https://repository.oca.argo.colossi.network/)

Variable q4 is an array(text) datatype. When generated as JSON from the repo it is interpreted to be an array. This makes parsing this result challenging because the datatype is array(text), it is not an array of the word text.

Is this an error? Should the syntax of OCA File be updated to not create this?

![image](https://github.com/user-attachments/assets/063c418a-6fb6-410e-8ddd-57fe229dedcf)

**Comment by blelump** [comment]
JSON, by design, is a loosely typed representation and doesn't bring type-rich collection types. Therefore, the way we represent such types, `Collection<T>` in this case, is purely the OCA Bundle design decision of how to handle this. Whether this is `["Text"]`, `{"type": "array", t: "Text"}` or any other approach, it always requires a parser that will extract the underlying information out of the representation and enable you to benefit from it. 

Therefore, it's not an error; it's a design choice.

**Comment by carlyh-micb** [comment]
Why did you make the design choice that makes it look like an array?

**Reply by blelump** [reply]
Mechanical convenience and lexical (semantical) unambiguity.
